packages/transport-tcp/src/utils.ts

```diff
@@ -22,7 +22,7 @@ export function multiaddrToNetConfig (addr: Multiaddr, config: NetConfig = {}):
   }
 
   // tcp listening
-  return { ...addr.toOptions(), ...config }
+  return { ...config, ...addr.toOptions() }
```

Comment:
Ensure we take the target host/port from the multiaddr and not the passed options.